ccinput/packages/psi4.py

```diff
@@ -10,40 +10,41 @@
     get_angle,
     get_dihedral,
     get_npxyz,
+    warn,
 )
 from ccinput.calculation import Calculation
 from ccinput.constants import CalcType, ATOMIC_NUMBER, LOWERCASE_ATOMIC_SYMBOLS
-from ccinput.exceptions import UnimplementedError
+from ccinput.exceptions import UnimplementedError, InvalidParameter
 
 
 class Psi4Calculation:
-    TEMPLATE = """
+    TEMPLATE = """#{header}
+
+{memory_line}
+
 molecule {{
 {charge} {multiplicity}  
-{coordinates}
-}}
+{coordinates}}}
 
-{command_line}
-"""
+{command_line}"""
 
     CALC_TYPES = {
         CalcType.SP: ["sp", "energy"],
-        CalcType.OPT: ["optimize"],
+        CalcType.OPT: ["opt", "optimize"],
         # CalcType.CONSTR_OPT,
-        # CalcType.FREQ,
+        CalcType.FREQ: ["freq", "frequency", "frequencies"],
         # CalcType.TS,
-        # CalcType.OPTFREQ,
+        CalcType.OPTFREQ: ["Opt+Freq Calculation", "optfreq", "opt+freq"],
     }
```

Comment:
`CALC_TYPES` is more to define the keywords for each calculation type and be able to get them programmatically (and make your job easier), rather than to detect synonyms. Detecting the calculation type is already handled by `get_abs_type` in `utilities.py`. The calculation type is already as `CalcType` enum.